File: app/src/main/java/com/mdove/ourflag/utils/GsonProvider.java
package com.mdove.ourflag.utils;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.mdove.ourflag.utils.json.JSONArraySerializer;
import com.mdove.ourflag.utils.json.JSONObjectSerializer;

import org.json.JSONArray;
import org.json.JSONObject;

/**
 * Created by MDove on 2019/4/21.
 */
public class GsonProvider {

    private static final Gson sDefault = new GsonBuilder()
        .registerTypeAdapter(JSONObject.class, new JSONObjectSerializer())
        .registerTypeAdapter(JSONArray.class, new JSONArraySerializer())
        .create();

    public static Gson getDefaultGson() {
        return sDefault;
    }
}

File: base/src/main/java/com/mdove/android/base/gson/GsonProvider.kt
package com.mdove.android.base.gson

import com.google.gson.GsonBuilder
import com.mdove.android.base.gson.json.JSONArraySerializer
import com.mdove.android.base.gson.json.JSONObjectSerializer
import org.json.JSONArray
import org.json.JSONObject

/**
 * Created by MDove on 2019/4/21.
 */
object GsonProvider {
    val defaultGson = GsonBuilder()
            .registerTypeAdapter(JSONObject::class.java, JSONObjectSerializer())
            .registerTypeAdapter(JSONArray::class.java, JSONArraySerializer())
            .create()
}
